rest-api-spec/test/template/30_validate_search_template.yaml

```diff
@@ -0,0 +1,55 @@
+---
+"Indexed Template validate tests":
+
+  - do:
+      put_template:
+        id: "1"
+        body: { "template": { "query": { "match": { "text": "{{my_value}}" } }, "size": "{{my_size}}" } }
+  - match: { _id: "1" }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      validate_search_template:
+        body: { "id": "1", "params": { "my_value": "foo", "my_size": 20 } }
+
+  - match: { template_output.query.match.text: "foo" }
+  - match: { template_output.size: "20" }
+
+  - do:
+      validate_search_template:
+        body: { "id": "1", "params": { "my_value": "bar", "my_size": 100 } }
+
+  - match: { template_output.query.match.text: "bar" }
+  - match: { template_output.size: "100" }
```

Comment:
In the script the size parameter is specified as a string because `"size": {{my_size}}` is not valid json and the parser error on it before we get a chance to run the mustache engine. This is a known bug and something we need to fix separately.